packages/@aws-cdk/aws-iam/lib/principals.ts

```diff
@@ -46,8 +46,32 @@ export interface IPrincipal extends IGrantable {
    *
    * @returns true if the statement was added, false if the principal in
    * question does not have a policy document to add the statement to.
+   *
+   * @deprecated Use `addToIdentityPolicy` instead.
    */
   addToPolicy(statement: PolicyStatement): boolean;
+
+  /**
+   * Add to the policy of this principal.
+   */
+  addToIdentityPolicy(statement: PolicyStatement): AddToIdentityPolicyResult;
+}
+
+/**
+ * Result of calling `addToIdentityPolicy`
+ */
+export interface AddToIdentityPolicyResult {
+  /**
+   * Whether the statement was added to the identity's policies.
+   */
+  readonly statementAdded: boolean;
+
+  /**
+   * Dependable which allows depending on the policy change being applied
+   *
+   * @default - Required if `statementAdded` is true.
```

Comment:
If this is always required if `statementAdded` is `true` then why do we need `statementAdded`? We can rename this field to `addedBy` or `owner` and require that it is set when a statement was added.

Reply:
`owner` doesn't really cover the load of what I'm trying to do here, which is strictly offer a way to add a dependency and nothing else.

As for `statementAdded`, yeah maybe it's superfluous. On the other hand it does make for more readable consumer code to say:

```ts
if (result.statementAdded) { /* do thing */ }
```

Rather than:

```ts
if (result.policyDependable !== undefined) { /* do a thing */ }
```

Open to a better name for the dependable though...

Reply:
Damn I don't.... Can we make this "@experimental"?